### ISSUE TITLE
Overlapping port ranges cause Retz to fail on startup

### ISSUE THREAD

**Issue by tgpfeiffer** [retz/retz#74]
On my cluster with four nodes (all configured using the same port range 31000-32000), Retz fails to start with

```
2016-11-10 16:13:29.296 INFO  [Thread-5] i.g.retz.scheduler.RetzScheduler - Offer stock renewal: 4 offers available (0 declined from stock)
Exception in thread "Thread-5" java.lang.RuntimeException: Port range overlapping: 31000-32000 and 31000-32000
    at io.github.retz.scheduler.Resource.merge(Resource.java:53)
    at io.github.retz.scheduler.RetzScheduler.resourceOffers(RetzScheduler.java:180)
I1110 16:13:29.329097 18192 sched.cpp:2021] Asked to abort the driver
```

Receiving the same port range from every node is not a bug, as far as I can tell. I am not sure what the `Resource.merge()` method does; maybe it is better to leave the `ports` resource out of there?

**Comment by kuenishi** [comment]
~~https://github.com/retz/retz/commit/e71b7f1c41a668c6b7f44d84e9711119b52095cb should be the right fix, which is included in [0.0.31](https://github.com/retz/retz/releases/tag/0.0.31)~~ . This wasn't enough.

**Comment by kuenishi** [comment]
Discussion, `Database#findFit()` currently considers total number of CPUs and Memories; summation of all current offers. `Resource.merge()` was intended just to calculate that summation here - but it is also used to merge offers of a single slave. Essentially calculating summation of the _amount_ of resources to know the capacity and merging port ranges are totally different things.

This is also related to a question how do we efficiently fetch job candidates from database - currently just running over iteration on queued jobs.

**Comment by kuenishi** [comment]
For now I'm closing this issue. If this is reproduced at 0.0.31 or later, feel free to reopen this issue.

**Comment by tgpfeiffer** [comment]
The bug still exists, via a different execution path:

```
2016-11-14 11:11:52.016 INFO  [Thread-5] i.g.retz.scheduler.RetzScheduler - Offer stock renewal: 6 offers available (0 declined from stock)
Exception in thread "Thread-5" java.lang.RuntimeException: Port range overlapping: 31000-32000 and 31000-32000
    at io.github.retz.scheduler.Resource.merge(Resource.java:53)
    at io.github.retz.scheduler.NaivePlanner.plan(NaivePlanner.java:153)
    at io.github.retz.scheduler.RetzScheduler.handleAll(RetzScheduler.java:231)
    at io.github.retz.scheduler.RetzScheduler.resourceOffers(RetzScheduler.java:188)
I1114 11:11:52.061102  2307 sched.cpp:2021] Asked to abort the driver
```

**Comment by kuenishi** [comment]
[Fix (merely a workaround) for this](https://github.com/retz/retz/commit/684176d67084ad4ce93f26fff856990a436b7509) is  included in [0.0.32](https://github.com/retz/retz/releases/tag/0.0.32) as far as we tested two different environemnts. Hope this works...

**Comment by tgpfeiffer** [comment]
Confirmed that bug is fixed in 0.0.32.
